resource.address/tcp/src/main/java/org/kaazing/gateway/resource/address/tcp/TcpResourceAddressFactorySpi.java

```diff
@@ -103,27 +103,22 @@ protected void parseNamedOptions0(String location, ResourceOptions options,
     private InetSocketAddress parseBindAddress(Object bindAddress) {
         if (bindAddress instanceof InetSocketAddress) {
             return (InetSocketAddress) bindAddress;
-        }
-        else if (bindAddress instanceof String) {
-            String[] bindParts = ((String) bindAddress).split(":");
-            switch (bindParts.length) {
-            case 1:
-                // port only
-                return new InetSocketAddress(parseInt(bindParts[0]));
-            case 2:
-                // hostname, port
-                String hostname = bindParts[0];
-                int port = parseInt(bindParts[1]);
-                return new InetSocketAddress(hostname, port);
-            }
-            // otherwise (more than one ":" separator encountered)
-            Pattern pattern = Pattern.compile("(\\[{1}@[a-zA-Z0-9 :]*\\]{1}):([0-9]*)");
+        } else if (bindAddress instanceof String) {
+            String address = (String) bindAddress;
+            // host and port
+            Pattern pattern = Pattern.compile("(.*):([0-9]*)");
```

Comment:
Seems like "[::1" doesn't throw any exception
"192.168.1.1:" doesn't throw IllegalArgumentException as line no:112 throws number format exception

I suggest just catch Exception for most of the code and throw IllegalArgumentException with a generic message.